src/compiler/core.ts

```diff
@@ -1134,10 +1134,33 @@ namespace ts {
      */
     export const supportedTypeScriptExtensions = [".ts", ".tsx", ".d.ts"];
     export const supportedJavascriptExtensions = [".js", ".jsx"];
-    const allSupportedExtensions  = supportedTypeScriptExtensions.concat(supportedJavascriptExtensions);
+
+    export function getExtensionsForGoal(goal: ResolutionGoal): string[] {

```

Comment:
`ResolutionGoal` seems to have some similarities to the `ExtensionPriority` enum used as part of glob resolution.